e2e.groovy

```diff
@@ -0,0 +1,58 @@
+def cico_retries = 16
+def cico_retry_interval = 60
+def ci_git_repo = 'https://github.com/Yuggupta27/ceph-csi'
+def ci_git_branch = 'e2e-centosci'
+def ref = "master"
+
+node('cico-workspace') {
+
+	stage('checkout ci repository') {
+		git url: "${ci_git_repo}",
+			branch: "${ci_git_branch}",
+			changelog: false
+	}
+
+	stage('reserve bare-metal machine') {
+		def firstAttempt = true
+		retry(30) {
+			if (!firstAttempt) {
+				sleep(time: 5, unit: "MINUTES")
+			}
+			firstAttempt = false
+			cico = sh(
+				script: "cico node get -f value -c hostname -c comment --retry-count ${cico_retries} --retry-interval ${cico_retry_interval}",
+				returnStdout: true
+			).trim().tokenize(' ')
+			env.CICO_NODE = "${cico[0]}.ci.centos.org"
+			env.CICO_SSID = "${cico[1]}"
+		}
+	}
+	try {
+		stage('prepare bare-metal machine') {
+			if ("${ghprbPullId}".length() != 0) {
+				ref = "pull/${ghprbPullId}/head"
+			}
+			sh 'scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ./prepare.sh ./multi-node-k8s.sh root@${CICO_NODE}:'
+			sh "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} ./prepare.sh --workdir=/opt/build/go/src/github.com/ceph/ceph-csi --gitrepo=${ci_git_repo} --ref=master"
+		}
+
+		stage('deploy kubernetes') {
+			sh 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} ./multi-node-k8s.sh'
+		}
+
+		stage('deploy rook') {
+			//sh 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} virsh reboot k8s-vagrant-multi-node_node2; sleep 120'
+			sh 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} /opt/build/go/src/github.com/ceph/ceph-csi/scripts/rook.sh deploy'
+		}
+
+		stage('build') {
+			sh 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} "export GOPATH=/opt/build/go ; go test github.com/ceph/ceph-csi/e2e --deploy-timeout=10 -timeout=30m -v"'
```

Comment:
This stage is badly named. It should probably be called "e2e".

After #1113 has been merged, and this PR is rebased, you can replace the ssh command with:
```
ssh ... make -C /opt/build/go/src/github.com/ceph/ceph-csi containerized-build TARGET=e2e.test`
```
to build the `e2e.test` executable inside a container (no need to install build dependencies like golang, lib*-devel) that way.

To run the test, execute:
```
ssh ... /opt/build/go/src/github.com/ceph/ceph-csi/e2e.test --deploy-timeout=10 -timeout=30m -v
```

Unfortunately it will be required to have librados and other dependencies installed. But, I think these libraries get installed when installing libvirt/qemu. So try it out and see if the executable fails with an error :eyes:

Reply:
Sure. Thanks!